Client/game_sa/CHandlingEntrySA.cpp

```diff
@@ -19,47 +19,54 @@ extern CGameSA* pGame;
 CHandlingEntrySA::CHandlingEntrySA()
 {
     // Create a new interface and zero it
-    m_pHandlingSA = new tHandlingDataSA;
-    memset(m_pHandlingSA, 0, sizeof(tHandlingDataSA));
-    m_bDeleteInterface = true;
+    if (m_HandlingSA = std::make_unique<tHandlingDataSA>())
+    {
+        memset(m_HandlingSA.get(), 0, sizeof(tHandlingDataSA));
+    }
 }
 
-CHandlingEntrySA::CHandlingEntrySA(tHandlingDataSA* pOriginal)
+CHandlingEntrySA::CHandlingEntrySA(const tHandlingDataSA* const pOriginal)
 {
     // Store gta's pointer
-    m_pHandlingSA = nullptr;
-    m_bDeleteInterface = false;
-    memcpy(&m_Handling, pOriginal, sizeof(tHandlingDataSA));
-}
-
-CHandlingEntrySA::~CHandlingEntrySA()
-{
-    if (m_bDeleteInterface)
+    m_HandlingSA = nullptr;
+    if (pOriginal)
     {
-        SAFE_DELETE(m_pHandlingSA);
+        memcpy(&m_Handling, pOriginal, sizeof(tHandlingDataSA));
```

Comment:
You should use our own layers for memory functions, so please use MemCpy & MemSet